python/altrios/__init__.py

```diff
@@ -31,3 +31,228 @@ def __array__(self):
 setattr(Pyo3Vec2Wrapper, "__array__", __array__)
 setattr(Pyo3Vec3Wrapper, "__array__", __array__)
 setattr(Pyo3VecBoolWrapper, "__array__", __array__)
+
+# for param_path_list() method to identify something as a struct so that it
+# checks for sub-variables and sub-structs, it must be added to this list:
+ACCEPTED_RUST_STRUCTS = ['FuelConverter', 
+                         'FuelConverterState', 
+                         'FuelConverterStateHistoryVec',
+                         'Generator',
+                         'GeneratorState',
+                         'GeneratorStateHistoryVec',
+                         'ReversibleEnergyStorage',
+                         'ReversibleEnergyStorageState',
+                         'ReversibleEnergyStorageStateHistoryVec',
+                         'ElectricDrivetrain',
+                         'ElectricDrivetrainState',
+                         'ElectricDrivetrainStateHistoryVec',
+                         'Locomotive',
+                         'LocoParams',
+                         'ConventionalLoco',
+                         'HybridLoco',
+                         'BatteryElectricLoco',
+                         'DummyLoco',
+                         'LocomotiveState',
+                         'LocomotiveStateHistoryVec',
+                         'LocomotiveSimulation',
+                         'PowerTrace',
+                         'Consist',
+                         'ConsistState',
+                         'ConsistStateHistoryVec',
+                         'ConsistSimulation',
+                         'SpeedTrace',
+                         'SetSpeedTrainSim',
+                         'SpeedLimitTrainSim',
+                         'LinkIdx',
+                         'LinkIdxTime',
+                         'TimedLinkPath',
+                         'LinkPoint',
+                         'Link',
+                         'Elev',
+                         'Heading',
+                         'Location',
+                         'Network',
+                         'LinkPath',
+                         'SpeedSet',
+                         'InitTrainState',
+                         'TrainState',
+                         'TrainStateHistoryVec',
+                         'TrainConfig',
+                         'TrainType',
+                         'TrainParams',
+                         'RailVehicle',
+                         'TrainSimBuilder',
+                         'SpeedLimitTrainSimVec',
+                         'EstTimeNet',
+                         'Pyo3VecWrapper',
+                         'Pyo3Vec2Wrapper',
+                         'Pyo3Vec3Wrapper',
+                         'Pyo3VecBoolWrapper']
+
+def param_path_list(self, path = "", param_path_list = []) -> list[str]:
```

Comment:
I think it's better to call this `var_path_list` or `variable_path_list`.  I usually use `param` for things that the user specifies.   Please change and propagate if this also makes sense to you.

Reply:
@robinsteuteville , be sure to propagate this to fastsim